GaussianNLLLoss no_batch_dim docs and testing

george-qi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69783

Differential Revision: [D33200486](https://our.internmc.facebook.com/intern/diff/D33200486)

george-qi committed:
GaussianNLLLoss no_batch_dim docs and testing

[ghstack-poisoned]

george-qi committed:
Update on "GaussianNLLLoss no_batch_dim docs and testing"




[ghstack-poisoned]

george-qi committed:
Update on "GaussianNLLLoss no_batch_dim docs and testing"




[ghstack-poisoned]

george-qi committed:
Update on "GaussianNLLLoss no_batch_dim docs and testing"




[ghstack-poisoned]

george-qi commented:
@george-qi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33200486).

jbschlosser commented on torch/testing/_internal/common_modules.py:
```diff
@@ -227,6 +227,31 @@ def reference_fn(m, p, i, t, constructor_kwargs=constructor_kwargs):
+    for desc, constructor_kwargs in cases:
+        module_inputs.append(
+            ModuleInput(constructor_input=FunctionInput(**constructor_kwargs),
+                        forward_input=FunctionInput(make_input((5, 3)),
+                                                    make_target((5, 3)),
+                                                    make_input((5, 1)).abs()),
```
I think these input shapes have batch dims, no?

jbschlosser reviewed (commented):
One small testing question but otherwise looks nice

george-qi committed:
Update on "GaussianNLLLoss no_batch_dim docs and testing"


Differential Revision: [D33200486](https://our.internmc.facebook.com/intern/diff/D33200486)

[ghstack-poisoned]

george-qi commented:
@george-qi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33200486).

jbschlosser reviewed (approved):
LGTM!
